core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,233 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', 'editorContextService', 'alertsService',
+  'INTERACTION_SPECS', 'stateSolutionService', 'explorationStatesService',
+  'SolutionHelperService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'explorationContextService',
+  function(
+    $scope, $rootScope, $modal, editorContextService, alertsService,
+    INTERACTION_SPECS, stateSolutionService, explorationStatesService,
+    SolutionHelperService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, explorationContextService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.SolutionHelperService = SolutionHelperService;
```

Comment:
OK I think GitHub did something here :P 
This comment was earlier at stateInteractionIdService.